business_requirement_deliverable/models/business.py

```diff
@@ -0,0 +1,160 @@
+# -*- coding: utf-8 -*-
+from openerp import api, fields, models
+
+
+class BusinessRequirementResource(models.Model):
+    _name = "business.requirement.resource"
+    _description = "Business Requirement Resource"
+
+    sequence = fields.Integer('Sequence')
+    description = fields.Char('Description', required=True)
+    product_id = fields.Many2one(
+        comodel_name='product.product',
+        string='Product',
+        required=False
+    )
+    uom_id = fields.Many2one(
+        comodel_name='product.uom',
+        string='UoM',
+        required=True
+    )
+    qty = fields.Integer(
+        string='Quantity'
+    )
+    resource_type = fields.Selection(
+        selection=[('task', 'Task'), ('procurement', 'Procurement')],
+        string='Type',
+        required=True,
+        default='procurement'
+    )
+    user_id = fields.Many2one(
+        comodel_name='res.users',
+        string='Assign To',
+        ondelete='set null'
+    )
+    business_requirement_deliverable_id = fields.Many2one(
+        comodel_name='business.requirement.deliverable',
+        string='Business Requirement Deliverable',
+        ondelete='cascade'
+    )
+    unit_price = fields.Float(
+        string='Unit Price'
+    )
+    price_total = fields.Float(
+        compute='_get_price_total',
+        string='Subtotal'
+    )
+
+    @api.one
+    @api.depends('unit_price', 'qty')
+    def _get_price_total(self):
+        self.price_total = self.unit_price * self.qty
+
+    @api.one
+    @api.onchange('product_id')
+    def product_id_change(self):
+        description = ''
+        uom_id = False
+        unit_price = 0
+        product = self.product_id
+        if product:
+            description = product.name
+            uom_id = product.uom_id.id
+            unit_price = product.standard_price
+        self.description = description
+        self.uom_id = uom_id
+        self.unit_price = unit_price
+
+    @api.one
+    @api.onchange('resource_type')
+    def resource_type_change(self):
+        if self.resource_type == 'procurement':
+            self.user_id = False
+
+
+class BusinessRequirementDeliverable(models.Model):
+    _name = "business.requirement.deliverable"
+    _description = "Business Requirement Deliverable"
```

Comment:
This model represents budget lines rather than deliverables.
I would name it `business.requirement.estimation`.
The related requirement_id is probably a high level requirement, that can be conceptually mapped to a Deliverable. So you could  label `business_requirement_id` as the related "Deliverable/Requirement"